### PR TITLE
Show error messages in Publish project

### PR DESCRIPTION
**Pull request by aosyatnik**
### Purpose

Link to YouTrack:
[MAGN-7834](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-7834) Show error messages, upload status in customizer publish dialog.

All error messages are stored in `Resource.resx` file. Also I created `PublishResourceDictionary.xaml`, where we can store colors and other useful stuff for UI.

Results:
1) If there was not Name:
image

2) If there was not Description:
image

3) If user closed Authentication window or authentication failed.
image

4) During Uploading:
image

5) If server not found or something went wrong during upload:
image

Status color panel is always at the bottom of Publish window.
image

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [ ] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files

### Reviewers

@pboyer 

### FYIs

@lillismith  
@Racel

### CODE REVIEW

**Comment by Racel** [comment]
This looks great, @aosyatnik! 

@lillismith @pboyer - is there some color scheme/ ui styling we should be using or establishing? I noticed some inconsistencies with this dialog and the ones we are making for custom paths.

File: src/DynamoPublish/Resource.resx [hunk 1]
```diff
@@ -117,25 +117,76 @@
...
+    <value>Failed</value>
+  </data>
   <data name="PageErrorMessage" xml:space="preserve">
     <value>"Page" is not set. set. Please, set server page in configuration file.</value>
   </data>
   <data name="PortErrorMessage" xml:space="preserve">
     <value>"Port" is not set. Please, set port in configuration file.</value>
   </data>
+  <data name="ProvideAuthProviderMessage" xml:space="preserve">
+    <value>Couldn't connect to server with this authentication provider. Try to use another provider.</value>
```
Comment (pboyer) [inline src/DynamoPublish/Resource.resx]: Dynamo could not authenticate. Please report this message to the Dynamo team.

File: src/DynamoPublish/Resource.resx [hunk 2]
```diff
@@ -117,25 +117,76 @@
...
+  </data>
+  <data name="ProvideWorskspaceDescriptionMessage" xml:space="preserve">
+    <value>Please, provide workspace description.</value>
```
Comment (pboyer) [inline src/DynamoPublish/Resource.resx]: Please provide a description of the workspace.

File: src/DynamoPublish/Resource.resx [hunk 3]
```diff
@@ -117,25 +117,76 @@
...
+  </data>
+  <data name="ProvideWorskspaceNameMessage" xml:space="preserve">
+    <value>Please, provide workspace name.</value>
```
Comment (pboyer) [inline src/DynamoPublish/Resource.resx]: Please provide a name for your workspace.

File: src/DynamoPublish/Resource.resx [hunk 4]
```diff
@@ -117,25 +117,76 @@
...
+  </data>
+  <data name="PublishButtonTitle" xml:space="preserve">
+    <value>Publish</value>
+  </data>
+  <data name="PublishViewDefaultDescriptionTitle" xml:space="preserve">
+    <value>enter description here</value>
```
Comment (pboyer) [inline src/DynamoPublish/Resource.resx]: A description of your customizer.

File: src/DynamoPublish/Resource.resx [hunk 5]
```diff
@@ -117,25 +117,76 @@
...
+  </data>
+  <data name="PublishViewDefaultNameTitle" xml:space="preserve">
+    <value>defaults to file name</value>
```
Comment (pboyer) [inline src/DynamoPublish/Resource.resx]: Is this field necessary?

File: src/DynamoPublish/Resource.resx [hunk 6]
```diff
@@ -117,25 +117,76 @@
...
+  </data>
+  <data name="PublishViewDescriptionTitle" xml:space="preserve">
+    <value>Description:</value>
```
Comment (pboyer) [inline src/DynamoPublish/Resource.resx]: Let's omit the colon the titles here.

File: src/DynamoPublish/Resource.resx [hunk 7]
```diff
@@ -117,25 +117,76 @@
...
+  </data>
+  <data name="PublishViewShareLinkTitle" xml:space="preserve">
+    <value>Share Link:</value>
+  </data>
   <data name="PublishViewTitle" xml:space="preserve">
     <value>Publish Customizer</value>
   </data>
-  <data name="ServerErrorMessage" xml:space="preserve">
+  <data name="ReadyForPublishMessage" xml:space="preserve">
+    <value>Workspace is ready for publish.</value>
```
Comment (pboyer) [inline src/DynamoPublish/Resource.resx]: Your workspace is ready to be published.

File: src/DynamoPublish/Resource.resx [hunk 8]
```diff
@@ -117,25 +117,76 @@
...
+  </data>
+  <data name="ServerNotFoundMessage" xml:space="preserve">
     <value>Server not found.</value>
   </data>
+  <data name="UnknownServerErrorMessage" xml:space="preserve">
+    <value>Error on the server side. Please, connect dynamo team for more information.</value>
```
Comment (pboyer) [inline src/DynamoPublish/Resource.resx]: There was an error when trying to publish your workspace. Please report this message to the Dynamo team.

**Comment by pboyer** [comment]
@Racel Yes, here are some dialogs:

### New custom node

newcustom

### New preset

preset

### Publish package (the basis of this work)

ready

**Comment by pboyer** [comment]
@aosyatnik I've made some suggestions on the text.

Can I also ask that you use some of the styles established above? 

In particular, can you:

1. Use `SDarkTextBox` for the fields
2. Match the color and size of the labels with the above dialogs (`Height="28" Foreground="DarkGrey"`)
3. Use `STextButton` for the buttons

**Comment by aosyatnik** [comment]
Thank you, @pboyer , for tips. I found styles, that you mentioned. That's how PublishView looks like now:

image

@Racel , you think this view you will like more :)

**Comment by Racel** [comment]
@aosyatnik - Looks better! Although, I am having trouble figuring out at a glance, the button hierarchy. If "Publish" is the "action button in this dialog, it is confusing that it is not at the bottom of dialog like the examples from @pboyer. To fix this, I suggest we add a horizontal break line above the 'Share Link' area to separate the publish and share features. 

Also, what does the 'Close' button do? Does it close the dialog or close the share link. It is kind of confusing. If it is to close the dialog then here is what I suggest: just place a 'Cancel' button beside publish as seen in the above examples and get rid of 'Close'.

**Comment by pboyer** [comment]
@Racel When fixing the merge conflict, I actually removed the close button, which wasn't really doing much.

**Comment by pboyer** [comment]
@Racel I think we have some bigger issues with the publish process, though. Like, how does the user understand what's going on at a high level? I'd love it if, for example, we had a diagram explaining what the actual process is and the limitations on it. I'd love to have a conversation about how to represent this and also rev this dialog while doing so.

**Comment by Racel** [comment]
@pboyer - sure thing. let me know when you want to discuss. @lillismith and I had some discussions on this a while ago. I think the plan was to implement something quickly right now, and refine as we go along.

**Pull request merged**